docs/eBpfExtensions.md

```diff
@@ -138,6 +138,84 @@ The extension passes a pointer to `context` inside `sample_program_context_heade
 `sample_program_context_header_t` when invoking the eBPF program. The header is not accessible
 by the program.
 
+##### Pointer members in program contexts
+
+The sample program context above contains two pointer fields, `data_start` and `data_end`.
+If the intent of an extension is to provide compatibility with some program type that exists on Linux,
+and pointer members exist in the program context, there is an inherent problem.
+
+On Linux, pointers in program contexts are always 32-bit, even on 64-bit platforms (conversion
```

Comment:
How did you figure this out? BTF works off of the assumption that pointers are 8 bytes. I'd be surprised if this was true in general.

Reply:
It is probably specific to those two structs (maybe others) but the point is to call out that implementers need to pay attention to this.

Reply:
@lmb I updated text